src/createAction.js

```diff
@@ -9,10 +9,14 @@ export default function createAction(type, actionCreator, metaCreator) {
 
   return (...args) => {
     const action = {
-      type,
-      payload: finalActionCreator(...args)
+      type
     };
 
+    const payload = finalActionCreator(...args);
+    if (!(payload === null || payload === undefined)) {
```

Comment:
Sorry, but it can be actually simplified with `if (payload != null)`. This will also return `false` for `null` or `undefined` only.